designer/apps/api/v1/views.py

```diff
@@ -41,3 +46,53 @@ def get_queryset(self):
         queryset = queryset.public().live()
 
         return queryset
+
+class ProgramDetailView(APIView):
+    """
+        Returns any portal related details for any program passed in through the `programs` query param
+
+        Input:
+            ?programs=<uuid_1>,<uuid_2>,<uuid_3>
+        Response:
+            [] => If not programs specified
+            400 => If uuids aren't valid
+            404 => If uuids don't exist in application
+            List of program detail objects => If all params are valid
+    """
+    permission_classes = (AllowAny, )
+
+    def get(self, *args, **kwargs):
+
+        requested_programs = self.request.query_params.get('programs')
+        if not requested_programs:
+            return Response([])
+        try:
+            program_uuids = [uuid.UUID(program_uuid) for program_uuid in requested_programs.split(',')]
+        except ValueError:
+            raise serializers.ValidationError('"programs" query param contains malformed UUIDs')
+
+        all_programs = []
+
+        for program_uuid in program_uuids:
+            try:
+                program_page = ProgramPage.objects.get(uuid=program_uuid)
+            except ProgramPage.DoesNotExist:
+                raise exceptions.NotFound()
+
+            frontend_url = generate_frontend_url(self.request, program_page)
+
+            all_programs.append({
+                'uuid': program_uuid,
```

Comment:
Can we add program_page.program_name here as well?

Reply:
Consider it done! Because it's done.